pkg/spooledtempfile/spooled_memory.go

```diff
@@ -0,0 +1,170 @@
+package spooledtempfile
+
+import (
+	"bufio"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+)
+
+// getSystemMemoryUsedFraction returns used/limit for the container if
+// cgroup limits are set; otherwise falls back to host /proc/meminfo.
+var getSystemMemoryUsedFraction = func() (float64, error) {
+	// 1) Try cgroup v2 (unified hierarchy)
+	if frac, ok, err := cgroupV2UsedFraction(); err != nil {
+		return 0, err
+	} else if ok {
+		return frac, nil
+	}
+
+	// 2) Try cgroup v1 (legacy)
+	if frac, ok, err := cgroupV1UsedFraction(); err != nil {
+		return 0, err
+	} else if ok {
+		return frac, nil
+	}
+
+	// 3) Fallback to host view
+	return hostMeminfoUsedFraction()
+}
+
+func cgroupV2UsedFraction() (frac float64, ok bool, err error) {
+	// Common modern path with systemd/docker/containerd/Nomad on cgroup v2
+	const (
+		usagePath = "/sys/fs/cgroup/memory.current"
+		limitPath = "/sys/fs/cgroup/memory.max"
```

Comment:
`memory.max` is the final safety net meaning that any RAM reading over this value will result in the OOM killer being invoked in the cgroup.
I think it would be a good idea to try and use `memory.high` if set